settings.yaml

```diff
@@ -6,16 +6,63 @@ logo_url:
   default: ""
 mobile_logo_url:
   default: ""
+
 links:
-  type: list
-  list_type: simple
-  default: ""
-  description: "Format: name,url,target (optional)"
+  type: objects
+  default: []
+  schema:
+    name: link
+    properties:
+      text:
+        type: string
+        required: true
+        validations:
+          min: 1
+          max: 100
+      url:
+        type: string
+        required: true
+        validations:
+          min: 1
+          max: 2048
+          url: true
+      target:
+        type: enum
+        default: _blank
+        choices:
+          - _blank
+          - _self
+          - _parent
+          - _top
+
 icons:
-  type: list
-  list_type: simple
-  default: ""
-  description: "Format: icon-name,url,target (optional)"
+  type: objects
+  default: []
+  schema:
+    name: icon
+    properties:
+      icon_name:
+        type: string
+        required: true
+        validations:
+          min: 1
+          max: 100
+      url:
+        type: string
+        required: true
+        validations:
+          min: 1
+          max: 2048
+          url: true
+      target:
+        type: enum
+        default: _blank
+        choices:
+          - _blank
```

Comment:
Btw, why the `_`?

Reply:
In `discourse-custom-header-links` it doesn't just use the target directly in the link's `target` attribute. 

https://github.com/discourse/discourse-custom-header-links/blob/6bd25fa7d740bd4888ddd8af3802feecc6e19bc2/javascripts/discourse/components/custom-header-links.js#L28

In the case of this plugin, the value of the property is used directly in the `target` attribute so we need to use the [actual valid values](https://www.w3schools.com/tags/att_a_target.asp)